src/Raven.Server/Documents/Indexes/VectorSearch/GenerateEmbeddings.cs

```diff
@@ -100,11 +132,55 @@ private static VectorValue Quantize(ByteStringContext allocator, in VectorEmbedd
         return embeddings;
     }
 
-    private static (IDisposable MemoryScope, Memory<byte> Memory, int UsedBytes) CreateEmbeddingViaSmartComponentsLocalEmbedding<TEmbedding>(ByteStringContext allocator, in string text, in int dimensions)
-        where TEmbedding : IEmbedding<TEmbedding>
+    private static (IDisposable MemoryScope, Memory<byte> Memory, int UsedBytes) CreateEmbeddingViaSmartComponentsLocalEmbedding(ByteStringContext allocator, in string text, in int dimensions)
     {
-        var memoryScope = allocator.Allocate(dimensions, out System.Memory<byte> memory);
-        Embedder.Embed<TEmbedding>(text, memory);
-        return (memoryScope, memory, dimensions);
+        List ??= new List<string>(1);
+        List.Add(text);
+
+        try
+        {
+            var embeddings = Embedder.Value.GenerateEmbeddingsAsync(List).GetAwaiter().GetResult();
+
+            var embedding = embeddings[0];
+
+            var memoryScope = allocator.Allocate(dimensions, out System.Memory<byte> memory);
+            
+            MemoryMarshal.AsBytes(embedding.Span).CopyTo(memory.Span);
+
+            return (memoryScope, memory, dimensions);
+        }
+        finally
+        {
+            List.Clear();
+        }
+    }
+
+    private static BertOnnxTextEmbeddingGenerationService CreateTextEmbeddingGenerationService()
+    {
+        using (var onnxModelStream = File.OpenRead(Path.Combine("LocalEmbeddings", "bge-micro-v2", "model.onnx")))
+        using (var vocabStream = File.OpenRead(Path.Combine("LocalEmbeddings", "bge-micro-v2", "vocab.txt")))
+        {
+            if (onnxModelStream == null)
+                throw new ArgumentNullException(nameof(onnxModelStream));
+            if (vocabStream == null)
+                throw new ArgumentNullException(nameof(vocabStream));
+
+            var options = new BertOnnxOptions();
+
+            var modelBytes = new MemoryStream();
+            onnxModelStream.CopyTo(modelBytes);
+
+            var onnxSession = new InferenceSession(modelBytes.Length == modelBytes.GetBuffer().Length ? modelBytes.GetBuffer() : modelBytes.ToArray(), OnnxSessionOptions ?? new SessionOptions());
```

Comment:
using?

Reply:
no, the BertOnnxTextEmbeddingGenerationService is using this